### PR TITLE
Create StreamLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Add StreamLayerClient skeleton
* Add unit test

Resolves: OLPEDGE-1521
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/StreamLayerClient.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export class StreamLayerClient {
+    /**
+     * Creates the [[StreamLayerClient]] instance.
+     *
+     * @param catalogHrn The HERE Resource Name (HRN) of the catalog from which you want to get partitions metadata and data.
+     * @param layerId The ID of the layer.
+     * @param settings The [[OlpClientSettings]] instance.
+     * @return The [[StreamLayerClient]] instance.
+     */
+    constructor(
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/StreamLayerClient.ts]: Please use named param StreamLayerClientParams

File: @here/olp-sdk-dataservice-read/test/unit/StreamLayerClient.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        streamLayerClient = new dataServiceRead.StreamLayerClient(
+            mockedHRN,
+            mockedLayerId,
+            (settings as unknown) as dataServiceRead.OlpClientSettings
+        );
+    });
+
+    it("Shoud be initialised", async () => {
+        assert.isDefined(streamLayerClient);
+        expect(streamLayerClient).be.instanceOf(
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/test/unit/StreamLayerClient.test.ts]: Please also check  if class properties initialized right

File: @here/olp-sdk-dataservice-read/lib/client/StreamLayerClient.ts (hunk 1 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/client/StreamLayerClient.ts]: Done

File: @here/olp-sdk-dataservice-read/test/unit/StreamLayerClient.test.ts (hunk 2 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/test/unit/StreamLayerClient.test.ts]: Done

**Pull request merged**